### DISCUSSION TITLE
[Solved] How to name the library

### DISCUSSION THREAD

**Discussion by Mikolaj** [Mikolaj/mostly-harmless#15] [General]
I'd like to create a new repo for the library, set up CI and move over only a subset of tests that run under 8GB and in reasonable time and exclude from there the MNIST data sets and the benchmarks, especially the comparative ones, with lots of code written using other libraries. Then, in this repo we could continue mad science experiments and brainstorming not restricted to the matters of the library. An alternative is to have several packages in one cabal project in this here repo, one package being the library, another (never on Hackage) a set of benchmarks. However, people are used to the repo named after the library and, in general, navigating a repo with many packages is a bit harder.

Another motivation: it's tiring to call it "the other library" and other similar names. ;)

Initial bets: heureka, dualdelta, let-ad. Also, does the OCaml library have a name? If so, we could stick 'h' in front of it.

**Comment by Mikolaj** [comment]
Today we agreed on something. Was it `horde-ad` as in "Higer Order Reverse DErivatives --- automatic differentiation"?

Edit. Yes, that's what I see in the google documents from the meeting. I will create the repo and tidy up the code and write a README.

**Reply by tomjaguarpaw** [reply]
And E can stand for "efficiently"

**Comment by simonpj** [comment]
Yes -- `horde-ad`.

**Comment by Mikolaj** [comment]
Edit: the discussion of the wiki document has moved to https://github.com/Mikolaj/horde-ad/issues/4

I forgot to ask for your preference regarding the license. However, this can be easily changed later on as well. For now I've gone with my default choice that liberates the code the most: GNU Affero General Public License v3.0.

The library repo is at 

https://github.com/Mikolaj/horde-ad

I've copied the `Delta` extension sketch from the Google doc to 

https://github.com/Mikolaj/horde-ad/wiki/Extending-Delta-to-primitive-Vector-type

which can be edited collaboratively once you accept the invite to the repo.

However, I noticed one can't conveniently discuss the document on the Wiki, so we may in the future reconsider and move it to a new discussion at 

https://github.com/Mikolaj/mostly-harmless/discussions

where the first post in the discussion would be the collaboratively edited document (I think this can be set up with proper user permissions) and all the other posts, laid out in a discussion tree, would be comments. As soon as the document is finalized we could either commit it to the `horde-ad` git repo or move it to the wiki. Let's experiment with the wiki first?

**Comment by simonpj** [comment]
> I forgot to ask for your preference regarding the license. However, this can be easily changed later on as well. For now I've gone with my default choice that liberates the code the most: GNU Affero General Public License v3.0.

Thanks.  I don't feel strongly, but I've always used the more liberal BSD license.

**Reply by goldfirere** [reply]
I second a preference for a more liberal license. BSD is fine.

**Reply by Mikolaj** [reply]
Thank you, noted. I've opened a ticket to change the license.

**Reply by Mikolaj** [reply]
Relicensed. Thank you for voicing your preference.
